internal/services/monitor/monitor_workbook_resource.go

```diff
@@ -0,0 +1,249 @@
+package monitor
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/suppress"
+
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/identity"
+
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/azure"
+	workbook "github.com/hashicorp/terraform-provider-azurerm/internal/services/monitor/sdk/2022-04-01/insights"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tags"
+
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
+	tagsHelper "github.com/hashicorp/go-azure-helpers/resourcemanager/tags"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/tf"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+func resourceMonitorWorkbook() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Create: resourceMonitorWorkbookCreateOrUpdate,
+		Read:   resourceMonitorWorkbookRead,
+		Update: resourceMonitorWorkbookCreateOrUpdate,
+		Delete: resourceMonitorWorkbookDelete,
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Update: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
+			_, err := workbook.ParseWorkbookID(id)
+			return err
+		}),
+
+		Schema: map[string]*pluginsdk.Schema{
+			"name": {
+				Type:             pluginsdk.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				DiffSuppressFunc: suppress.CaseDifference,
+				ValidateFunc:     validation.IsUUID,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"location": azure.SchemaLocation(),
+
+			"display_name": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"serialized_data": {
+				Type:             pluginsdk.TypeString,
+				Required:         true,
+				ValidateFunc:     validation.StringIsJSON,
+				DiffSuppressFunc: pluginsdk.SuppressJsonDiff,
+			},
+
+			"source_id": {
+				Type:             pluginsdk.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				DiffSuppressFunc: suppress.CaseDifference,
+				ValidateFunc:     validation.StringIsNotEmpty,
+			},
+
+			"category": {
+				Type:         pluginsdk.TypeString,
+				Optional:     true,
+				Default:      "workbook",
+				ValidateFunc: validation.StringIsNotEmpty,
```

Comment:
Presumably there's known possible values for this we can validate against? Also we should be using Title Case here?

Reply:
1. There are 2 known values and I have added them into the resource document. However, it can also be a customized value defined by users so that it cannot be defined as an enumeration string.
2. "workbook" is a known case-sensitive value defined by Azure. That's why I did not use Title Case here.